src/nunit.analyzers.tests/IgnoreCaseUsage/IgnoreCaseUsageAnalyzerTests.cs

```diff
@@ -0,0 +1,83 @@
+using Gu.Roslyn.Asserts;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NUnit.Analyzers.IgnoreCaseUsage;
+using NUnit.Framework;
+
+namespace NUnit.Analyzers.Tests.IgnoreCaseUsage
+{
+    public class IgnoreCaseUsageAnalyzerTests
+    {
+        private static readonly DiagnosticAnalyzer analyzer = new IgnoreCaseUsageAnalyzer();
+
+        [Test]
+        public void AnalyzeWhenIgnoreCaseUsedForNonStringEqualToArgument()
+        {
+            var testCode = TestUtility.WrapInTestMethod(@"
+                Assert.That(1, Is.EqualTo(1).↓IgnoreCase);");
+
+            AnalyzerAssert.Diagnostics(analyzer, testCode);
+        }
+
+        [TestCase(nameof(Is.EqualTo))]
+        [TestCase(nameof(Is.EquivalentTo))]
+        [TestCase(nameof(Is.SubsetOf))]
+        [TestCase(nameof(Is.SupersetOf))]
+        public void AnalyzeWhenIgnoreCaseUsedForNonStringCollection(string constraintMethod)
+        {
+            var testCode = TestUtility.WrapInTestMethod($@"
+                var actual = new[] {{1,2,3}};
+                var expected = new[] {{3,2,1}};
+                Assert.That(actual, Is.{constraintMethod}(expected).↓IgnoreCase);");
+
+            AnalyzerAssert.Diagnostics(analyzer, testCode);
+        }
+
+        [Test]
+        public void ValidWhenIgnoreCaseUsedForStringEqualToArgument()
+        {
+            var testCode = TestUtility.WrapInTestMethod(@"
+                Assert.That(""A"", Is.EqualTo(""a"").IgnoreCase);");
+
+            AnalyzerAssert.NoAnalyzerDiagnostics(analyzer, testCode);
+        }
+
+        [TestCase(nameof(Is.EqualTo))]
+        [TestCase(nameof(Is.EquivalentTo))]
+        [TestCase(nameof(Is.SubsetOf))]
+        [TestCase(nameof(Is.SupersetOf))]
+        public void ValidWhenIgnoreCaseUsedForStringCollection(string constraintMethod)
+        {
+            var testCode = TestUtility.WrapInTestMethod($@"
+                var actual = new[] {{""a"",""b"",""c""}};
+                var expected = new[] {{""A"",""C"",""B""}};
+                Assert.That(actual, Is.{constraintMethod}(expected).IgnoreCase);");
+
+            AnalyzerAssert.NoAnalyzerDiagnostics(analyzer, testCode);
+        }
+
+        [Test]
+        public void ValidWhenNonGenericIEnumerableProvided()
+        {
+            var testCode = TestUtility.WrapInTestMethod(@"
+                var actual = new[] {""a"",""b"",""c""};
+                System.Collections.IEnumerable expected = new[] {""A"",""C"",""B""};
+                Assert.That(actual, Is.EqualTo(expected).IgnoreCase);");
+
+            AnalyzerAssert.NoAnalyzerDiagnostics(analyzer, testCode);
+        }
+
+        [Test]
+        public void ValidWhenStringDictionaryProvided()
```

Comment:
That one I wasn't sure about. 
That's how it works right now, but looking at NUnit code I wasn't sure if it's actually intended behavior.

Split tests accordind to your suggestion anyway.

Reply:
I took a closer look at the comparers in NUnit (in the folder https://github.com/nunit/nunit/tree/master/src/NUnitFramework/framework/Constraints/Comparers), and the intent is to pass the `IgnoreCase` to the underlying constituents for types that contain underlying types of type string and char (expect to the key when comparing dictionaries). So the following makes sense (albeit a little contrived) and succeeds.

```csharp
        [Test]
        public void TestNested()
        {
            var actual = new[] { new[] { new[] { "Test" } } };
            var expected = new[] { new[] { new[] { "TEST" } } };
            Assert.That(actual, Is.EqualTo(expected).IgnoreCase);
        }
```

So actually we should also handle that the `IgnoreCase` makes sense for :
* Comparison of Tuple and ValueTuple (currently there is an error, so that this only work as intended on master)
* Comparison of DictionaryEntry (both on key and value)
* Comparison of KeyValuePair (both on key and value
* and handle that the nesting can be more than one level (will probably not happen that often)

I'm happy about the current state of this PR, and we can always improve on it later on. I'll let it be up to you whether we should merge the PR now or whether we should try to handle more of the cases. Personally, I think the current code covers most of the regular use-cases.